### ISSUE TITLE
Unable to find element by xpath, id, name, class name, -ios predicate string, -ios class chain, accessibility id

### ISSUE THREAD

**Issue by vishalpaliwal1** [appium/appium#13846]
## The problem

trying to get elements using xpath or accessibility label...but appium is giving error..unable to find ...

## Environment

* Appium version (or git revision) that exhibits the issue: 1.16.0
* Last Appium version that did not exhibit the issue (if applicable): NA
* Desktop OS/version used to run Appium: MACOS 10.15.2 (19C57)
* Node.js version (unless using Appium.app|exe): v13.6.0
* Npm or Yarn package manager: 6.13.4
* Mobile platform/version under test: IOS 13.xx 
* Real device or emulator/simulator: Simulator
* Appium CLI or Appium.app|exe: Appium CLI

## Details

If necessary, describe the problem you have been experiencing in more detail.

## Link to Appium logs
Full logs:-
https://gist.github.com/vishalpaliwal1/ecc6d64d8921327374b67a1d280a7f0d

main error log:
method 1
[debug] [W3C (6fc246dd)] Calling AppiumDriver.findElements() with args: ["xpath","//*[@Name=\"Sign_in\"]","6fc246dd-a43c-4eec-b29e-4eab4ac1c73e"]
[debug] [WD Proxy] Got response with status 200: {"value":[],"sessionId":"9358A917-8775-4F6B-9785-D9F2838B4F59"}
[debug] [W3C (6fc246dd)] Responding to client with driver.findElements() result: []

method 2
[HTTP] {"using":"accessibility id","value":"Sign_in"}
[debug] [W3C (72c86a74)] Calling AppiumDriver.findElement() with args: ["accessibility id","Sign_in","72c86a74-c028-4fbe-ad4d-bb9c145067d4"]
[debug] [XCUITest] Executing command 'findElement'
[debug] [BaseDriver] Valid locator strategies for this request: xpath, id, name, class name, -ios predicate string, -ios class chain, accessibility id
[debug] [BaseDriver] Waiting up to 0 ms for condition
[debug] [WD Proxy] Matched '/element' to command name 'findElement'
[debug] [WD Proxy] Proxying [POST /element] to [POST http://localhost:8100/session/E01AC91C-795F-48C7-8433-475157B3AFF7/element] with body: {"using":"accessibility id","value":"Sign_in"}
[WD Proxy] Got an unexpected response with status 404: {"value":{"error":"no such element","message":"unable to find an element using 'accessibility id', value 'Sign_in'","traceback":"(\n\t0   WebDriverAgentLib                   0x000000010c88632d FBNoSuchElementErrorResponseForRequest + 301\n\t1   WebDriverAgentLib                   0x000000010c8860d9 +[FBFindElementCommands handleFindElement:] + 425\n\t2   WebDriverAgentLib                   0x000000010c862b06 -[FBRoute_TargetAction mountRequest:intoResponse:] + 182\n\t3   WebDriverAgentLib                   0x000000010c84e18b __37-[FBWebServer registerRouteHandlers:]_block_invoke + 555\n\t4   RoutingHTTPServer                   0x000000010c951ac7 -[RoutingHTTPServer handleRoute:withRequest:response:] + 136\n\t5   RoutingHTTPServer                   0x000000010c9523da __72-[RoutingHTTPServer routeMethod:withPath:parameters:request:connection:]_block_invoke + 47\n\t6   libdispatch.dylib                   0x00007fff5223f7b9 _dispatch_client_callout + 8\n\t7   libdispatch.dylib                   0x00007fff5224c...
[debug] [W3C] Matched W3C error code 'no such element' to NoSuchElementError

## Code To Reproduce Issue [ Good To Have ]

@@ -38,6 +38,8 @@ class YourUltimateContent extends React.PureComponent {
           onPress={() => this.navigate('Login')}
           titleColor={this.props.themeColors.OnBoarding_YourUltimateContent_Button_Signin_textColor}
           style={[styles.button, { marginBottom: 68 / 2, }, buttons.materialButton.signinButton]}
+          accessible={true}
+          accessibilityLabel="Sign_in"
         />
         <LargeWhiteOnBlack80Btn
           buttonText={I18n.t('OnBoarding.CreateAccount')}

basic_test.py::BasicTest::test_basic_test FAILED

========================================================================================== FAILURES ===========================================================================================
__________________________________________________________________________________ BasicTest.test_basic_test __________________________________________________________________________________

self = <basic_test.BasicTest testMethod=test_basic_test>

        def test_basic_test(self):
            set_up(self)
>   	el1 = self.driver.find_element_by_accessibility_id("Sign_in")

basic_test.py:11: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python2.7/site-packages/appium/webdriver/extensions/search_context/mobile.py:43: in find_element_by_accessibility_id
    return self.find_element(by=MobileBy.ACCESSIBILITY_ID, value=accessibility_id)
/usr/local/lib/python2.7/site-packages/appium/webdriver/webdriver.py:284: in find_element
    'value': value})['value']
/usr/local/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py:321: in execute
    self.error_handler.check_response(response)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <appium.webdriver.errorhandler.MobileErrorHandler object at 0x10558d590>
response = {'status': 404, 'value': '{"value":{"error":"no such element","message":"An element could not be located on the page u....doNativeFind (/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/lib/commands/find.js:126:11)"}}'}

    def check_response(self, response):
        try:
            super(MobileErrorHandler, self).check_response(response)
        except WebDriverException as wde:
            if wde.msg == 'No such context found.':
                raise NoSuchContextException(wde.msg, wde.screen, wde.stacktrace)
            else:
>               raise wde
E               NoSuchElementException: Message: An element could not be located on the page using the given search parameters.

/usr/local/lib/python2.7/site-packages/appium/webdriver/errorhandler.py:29: NoSuchElementException
================================================================================== 1 failed in 16.11 seconds ==================================================================================

**Comment by mykola-mokhnach** [comment]
Is the element present in the page source output? Would you mind attaching this output?

**Comment by vishalpaliwal1** [comment]
Can see elements on Appium-desktop..but same elements cant be found using appium/codeceptjs script s..also python script....same issue....

```from appium import webdriver
import os

def set_up(self):
    d_caps = {}
    d_caps['platformName'] = 'iOS'
    d_caps['platformVersion'] = '13.3'
    d_caps['deviceName'] = 'iPhone 11'
    d_caps['automationName'] = 'XCUITest'
    d_caps['locationServicesEnabled'] = 'true'
    d_caps['locationServicesAuthorized'] = 'true'
    d_caps['calendarAccessAuthorized'] = 'true'
    d_caps['simpleIsVisibleCheck'] = 'true'
    d_caps['fullReset'] = 'false'
    d_caps['noReset'] = 'true'
    d_caps['newCommandTimeout'] = '3600'
    d_caps['app'] = os.path.abspath('xyz.app')
self.driver = webdriver.Remote('http://localhost:4723/wd/hub', d_caps)```


cat basic_test.py 
import unittest
import time
from common_utils import set_up

class BasicTest(unittest.TestCase):
    test_name = 'BasicTest'
    driver = None

    def test_basic_test(self):
        set_up(self)
	el1 = self.driver.find_element_by_accessibility_id("Sign_in")
	el1.click()

![image](https://user-images.githubusercontent.com/59428124/72508449-a4330d80-3880-11ea-8d3d-e51d78e5dc63.png)

**Comment by mykola-mokhnach** [comment]
I still would like to see the page source output

**Comment by amedvedjev** [comment]
@vishalpaliwal1 you should use
```
accessibilityIdentifier
```
in iOS code instead. Plus XCUITest has problem when some elements are not visible! (e.g. when we have elements inside button). Same may happen with cells. There are some workarounds for such cases - ask iOS developers.

**Comment by vishalpaliwal1** [comment]
I tried with Android also, behavoiur was same...below is html...
https://gist.github.com/vishalpaliwal1/7d9f5d76235ea4410678374544809420

**Comment by mykola-mokhnach** [comment]
This is the html page source. I want the xml page source generated by Appium

**Comment by vishalpaliwal1** [comment]
can you guide me to location ? I dnt knw whr to find it..

**Comment by mykola-mokhnach** [comment]
You could use Appium [forum](discuss.appium.io) to ask questions

**Comment by mykola-mokhnach** [comment]
Closed as not an issue.
